src/main/resources/application.properties

```diff
@@ -6,4 +6,7 @@ spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
 #enabling the H2 console
 spring.h2.console.enabled=true
 # Countries now Currency Url
-countriesnow.currency.url=https://countriesnow.space/api/v0.1/countries/currency
\ No newline at end of file
+countriesnow.currency.url=https://countriesnow.space/api/v0.1/countries/currency
+# News API Url
+newsapi.url=https://newsapi.org/v2/top-headlines?language={lang}&category={category}
```

Comment:
This is fine. But the issue is that it will send the parameters always to the newsAPI side.also we have to be aware of all the parameters(which is absolutely fine in our case). You can find an example here on how to build query parameters dynamically https://stackoverflow.com/a/54447041